src/utils.cpp

```diff
@@ -924,8 +924,17 @@ uint32_t kernel_version(int attempt)
 
 std::string abs_path(const std::string &rel_path)
 {
-  auto p = std_filesystem::path(rel_path);
-  return std_filesystem::canonical(std_filesystem::absolute(p)).string();
+  // filesystem::canonical does not work very well with /proc paths
+  // failing during canonicalization. See iovisor:bpftrace#1595
+  if (rel_path.rfind("/proc", 0) != 0)
```

Comment:
IIUC, the issue isn't that fs::canonical() doesn't work on procfs [0], it's that we're trying to canonicalize across mount namespaces. And part of the issue is that `/proc/pid/root` is a symlink to target pid's root as viewed by the target.

Wouldn't a better fix be to check for `/root` as well as `/proc`?

[0]: http://ix.io/2EQ0 prints `"/usr/bin"` for me

Reply:
Have clarified the comment, limited the check for `/proc/<pid>/root`, and improved the test to cover the functions behaviour.  

I assume by "test that exercises the previously broken behavior" you meant a test that actually tries to trace with a path of a process from another mount ns? That test already exists but is disabled, I have created an issue to re-enable it: #1638

Reply:
> I assume by "test that exercises the previously broken behavior" you meant a test that actually tries to trace with a path of a process from another mount ns? That test already exists but is disabled, I have created an issue to re-enable it: #1638

Nice :)